src/db.rs

```diff
@@ -98,6 +98,38 @@ impl DbInner {
         Ok(None)
     }
 
+    async fn fence_writers(&self, manifest: &mut FenceableManifest) -> Result<(), SlateDBError> {
+        let wal_id_last_compacted = self.state.read().state().core.last_compacted_wal_sst_id;
+        let max_wal_id = self
+            .table_store
```

Comment:
There are two things that I think are a not still ideal.
1. `FenceableManifest` has to be passed in to db_inner, we do need to check for new epoch and this is the natural way to do it. But previously db_inner didn't need manifest, now it does for this case.
2. `get_wal_sst_list` is called twice, I thought about doing it once and reusing the result in `replay_wal`, it causes more things to be passed around, left it as is, since this is only at startup.

Reply:
Yeah I'm not too bothered by (2). If it's a problem we can optimize it later. Let's optimize for readability right now. See below for my thoughts on (1).

Reply:
Resolving this, and using the other comment to track.